### PR TITLE
Add 4DForm as a JSON language

### PR DESCRIPTION
**Pull request by Ganbin**
## Description
4DForm extension contains JSON data representing 4D Form description. It is used in the context of 4D software application which start to provide source control compatibility in beta within the new v17 R6 version and will be released officially in v18 (between end 19 - begin 20). More and more repositories contains that kind of files.

## Checklist:

- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3A4DForm+pages+rightMargin+NOT+nothack
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - Example not on GitHub. It is a sample form that I have used during a training of 4D software development platform.
    - Sample license(s):
      - MIT
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2337,6 +2338,7 @@ JSON:
   - ".tern-project"
   - ".watchmanconfig"
   - composer.lock
+  - form.4DForm
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: This shouldn't be necessary as you already added `.4DForm` as a JSON extension.

Comment (Ganbin) [reply lib/linguist/languages.yml]: Ok I was not sure If I had to add it in the filename too.

**Comment by pchaigno** [comment]
@Alhadis Did you already check usage? Each repository seems to hold a lot of files.

**Comment by Ganbin** [comment]
Hi, thanks for the interest and the check. Effectively, the non-beta (v18) should be released in early 2020. We could expect an increasing number of directories in the mid-2020. I also think that a lot of repositories may be private and are therefore not measurable.

**Comment by pchaigno** [comment]
@Ganbin I usually check usage every ~6 months and stop monitoring file extensions for which usage hasn't grown. So it's maybe best if I start checking around the time the non-beta is released... Couldn't you post here when that happens?

**Comment by Ganbin** [comment]
@pchaigno Ok I will try to remember to post here in about 6 month when the non-beta will be released.

**Pull request closed without merging**